submissions/models.py

```diff
@@ -107,6 +109,98 @@ def from_db_value(self, value, expression, connection, context=None):
         return json.loads(value, **self.load_kwargs)
 
 
+# Has this submission been soft-deleted? This allows instructors to reset student
+# state on an item, while preserving the previous value for potential analytics use.
+DELETED = u'D'
+ACTIVE = u'A'
+STATUS_CHOICES = (
+    (DELETED, u'Deleted'),
+    (ACTIVE, u'Active'),
+)
+
+
+@python_2_unicode_compatible
+class TeamSubmission(TimeStampedModel):
+    """
+    A single response by a team for a given problem (ORA2) in a given course.
+    An abstraction layer over Submission used to for teams. Since we create a submission record for every team member,
+    there is a need to have a single point to connect the team to the workflows.
+    TeamSubmission is a many to many with Submission
+    """
+
+    uuid = models.UUIDField(db_index=True, default=uuid4, null=False)
+
+    # Which attempt is this? Consecutive Submissions do not necessarily have
+    # increasing attempt_number entries -- e.g. re-scoring a buggy problem.
+    attempt_number = models.PositiveIntegerField()
+
+    # submitted_at is separate from created_at to support re-scoring and other
+    # processes that might create Submission objects for past user actions.
+    submitted_at = models.DateTimeField(default=now, db_index=True)
+
+    # doosNOT need it    's own index if the UNIQUE constraint detailed below exists
+    # (with course_id as first field)
```

Comment:
Nit: you can delete this comment, since we decided we couldn't have that unique constraint.